master/internal/api_model.go

```diff
@@ -22,19 +22,54 @@ import (
 	structpb "github.com/golang/protobuf/ptypes/struct"
 )
 
-func (a *apiServer) GetModel(
-	_ context.Context, req *apiv1.GetModelRequest) (*apiv1.GetModelResponse, error) {
+func (a *apiServer) ModelFromIdentifier(identifier string) (*modelv1.Model, error) {
+	var err error
 	m := &modelv1.Model{}
-	switch err := a.m.db.QueryProto("get_model", m, req.ModelName); err {
+
+	allNumbers, _ := regexp.MatchString("^\\d+$", identifier)
+	if allNumbers {
+		err = a.m.db.QueryProto("get_model_by_id", m, identifier)
+	} else {
+		err = a.m.db.QueryProto("get_model", m, identifier)
+	}
+	switch err {
 	case db.ErrNotFound:
 		return nil, status.Errorf(
-			codes.NotFound, "model \"%s\" not found", req.ModelName)
+			codes.NotFound, "model \"%s\" not found", identifier)
+	default:
+		return m, errors.Wrapf(err,
+			"error fetching model \"%s\" from database", identifier)
+	}
+}
+
+func (a *apiServer) ModelVersionFromID(modelIdentifier string,
+	versionID int32) (*modelv1.ModelVersion, error) {
+	mv := &modelv1.ModelVersion{}
+	parentModel, err := a.ModelFromIdentifier(modelIdentifier)
+	if err != nil {
+		return nil, err
+	}
+
+	switch err = a.m.db.QueryProto(
+		"get_model_version", mv, parentModel.Id, versionID); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(
+			codes.NotFound, "model \"%s\" version %d not found", modelIdentifier, versionID)
 	default:
-		return &apiv1.GetModelResponse{Model: m},
-			errors.Wrapf(err, "error fetching model \"%s\" from database", req.ModelName)
+		return mv, errors.Wrapf(err,
+			"error fetching model \"%s\" version %d from database", modelIdentifier, versionID)
```

Comment:
suggestion: use `%q` instead of `\"%s\"` and `%v` instead of `%d`:
- `%q` is easier to read that `\"%s\"` and offers more correct quoting
- `%d` offers nothing that that `%v` doesn't offer, only `%d` is more strict (easier to mess up).

Another suggestion: reword to emphasize what was not found: `"error fetching version %v for model %q from database"`

Reply:
will do :+1: